### ISSUE TITLE
Suggestion: Croma.Nilable

### ISSUE THREAD

**Issue by ymtszw** [skirino/croma#2]
As discussed IRL, `use`-able `Croma.Nilable` type module generator would be nice to have.

It allows adding `default/0` via `:default` option, which currently cannot possible via `Croma.TypeGen.nilable/1` ad-hoc generator.

```ex
defmodule Id do
  use Croma.SubtypeOfString, pattern: ~r/id_\w+/
end

defmodule NilableId do
  use Croma.Nilable, elem_module: Id, default: nil
end

NilableId.default()
# => nil
```

The above `NilableId`, when used in `Croma.Struct`, can supply default value via its `default/0`.

```ex
defmodule SomeData do
  use Croma.Struct, fields: [id: NilableId]
end

SomeData.new([])
# => %SomeData{id: nil}
```

This is convenient when you want/need to have multiple nilable fields in `SomeData` struct.

**Comment by ymtszw** [comment]
Regarding to leverage `new/1` of `Croma.Struct`, same thing can be achieved just by adding `default/0` in modules generated by `Croma.TypeGen.nilable/1`.

Actually this might be better since `nilable/1` generator is already common and it doesn't require defining separate module for this purpose only.

**Comment by skirino** [comment]
Thank you for your suggestion!
I've published v0.6.0 release that addresses this issue by adding `default/0` in modules generated by `nilable/1` (and other adhoc generators).
